src/libs/actions/Policy.js

```diff
@@ -222,17 +223,91 @@ function removeMembers(members, policyID) {
     }, {optimisticData, failureData});
 }
 
+/**
+* Optimistically create a chat for each member of the workspace, creates both optimistic and success data for onyx.
+*
+* @param {String} policyID
+* @param {Array} members
+* @param {Array} betas
+* @returns {Object} - object with onyxSuccessData, onyxOptimisticData, and optimisticReportIDs (map login to reportID)
+*/
+function createPolicyExpenseChats(policyID, members, betas) {
+    const workspaceMembersChats = {
+        onyxSuccessData: [],
+        onyxOptimisticData: [],
+        optimisticReportIDs: {},
+    };
+
+    // If the user is not in the beta, we don't want to create any chats
+    if (!Permissions.canUsePolicyExpenseChat(betas)) {
+        return workspaceMembersChats;
+    }
+
+    _.each(members, (login) => {
+        const optimisticReport = ReportUtils.buildOptimisticChatReport(
+            [login],
+            undefined,
+            CONST.REPORT.CHAT_TYPE.POLICY_EXPENSE_CHAT,
+            policyID,
+            login,
+        );
+        const optimisticCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(optimisticReport.ownerEmail);
+
+        workspaceMembersChats.optimisticReportIDs[login] = optimisticReport.reportID;
+        workspaceMembersChats.onyxOptimisticData.push({
+            onyxMethod: CONST.ONYX.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticReport.reportID}`,
+            value: {
+                ...optimisticReport,
+                pendingFields: {
+                    createChat: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                },
+                isOptimisticReport: true,
+            },
+        });
+        workspaceMembersChats.onyxOptimisticData.push({
+            onyxMethod: CONST.ONYX.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticReport.reportID}`,
+            value: {[optimisticCreatedAction.reportActionID]: optimisticCreatedAction},
+        });
+        workspaceMembersChats.onyxSuccessData.push({
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticReport.reportID}`,
+            value: {
+                isLoadingReportActions: false,
```

Comment:
I was modelling that on creating report inside `Report.js` file in function `openReport` (393:400 lines)

Reply:
Looks like `false` is the default here https://github.com/Expensify/App/blob/64f4a7814e87bd7c4d085776abf5cb8a67b9e8bc/src/pages/home/ReportScreen.js#L96

and it's used here https://github.com/Expensify/App/blob/64f4a7814e87bd7c4d085776abf5cb8a67b9e8bc/src/pages/home/ReportScreen.js#L228

as well as here
```
src/pages/home/report/ReportActionsList.js
180:                        if (this.props.report.isLoadingReportActions && lastReportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.CREATED) {

src/pages/home/report/ReportActionsView.js
134:        if (nextProps.report.isLoadingReportActions !== this.props.report.isLoadingReportActions) {
```

Seems like it can probably be removed.